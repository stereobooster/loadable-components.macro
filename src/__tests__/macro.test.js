const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "no usage": `import loadableImport from "../macro";`,
    basic: `
      import loadableImport from "../macro";
      import loadable from "loadable-components";

      const AsyncComponent = loadable(() => loadableImport("./MyComponent"));
    `,
    "with options": `
      import loadableImport from "../macro";
      import loadable from "loadable-components";

      const AsyncComponent = loadable(() => loadableImport("./MyComponent"), {
        LoadingComponent: () => "Loading...",
      });
    `,
    "multiple imports": `
      import loadableImport from "../macro";
      import loadable from "loadable-components";

      const What = loadable(async () => {
        const { default: DeepWord } = await loadableImport("./DeepWorld");
        const { default: DeepAmazing } = await loadableImport("./DeepAmazing");
        return [DeepWord, DeepAmazing];
      });
    `,
  },
});
