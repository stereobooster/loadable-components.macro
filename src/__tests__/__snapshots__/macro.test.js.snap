// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros basic: basic 1`] = `
"
import loadableImport from \\"../macro\\";
import loadable from \\"loadable-components\\";

const AsyncComponent = loadable(() => loadableImport(\\"./MyComponent\\"));

      ↓ ↓ ↓ ↓ ↓ ↓

import loadable from \\"loadable-components\\";

const AsyncComponent = loadable(
  () => import(/*webpackChunkName: MyComponent*/ \\"./MyComponent\\"),
  {
    modules: [\\"MyComponent\\"],
  }
);

"
`;

exports[`macros multiple imports: multiple imports 1`] = `
"
import loadableImport from \\"../macro\\";
import loadable from \\"loadable-components\\";

const What = loadable(async () => {
  const { default: DeepWord } = await loadableImport(\\"./DeepWorld\\");
  const { default: DeepAmazing } = await loadableImport(\\"./DeepAmazing\\");
  return [DeepWord, DeepAmazing];
});

      ↓ ↓ ↓ ↓ ↓ ↓

import loadable from \\"loadable-components\\";

const What = loadable(
  async () => {
    const {
      default: DeepWord,
    } = await import(/*webpackChunkName: DeepWorld*/ \\"./DeepWorld\\");
    const {
      default: DeepAmazing,
    } = await import(/*webpackChunkName: DeepAmazing*/ \\"./DeepAmazing\\");
    return [DeepWord, DeepAmazing];
  },
  {
    modules: [\\"DeepWorld\\", \\"DeepAmazing\\"],
  }
);

"
`;

exports[`macros no usage: no usage 1`] = `
"
import loadableImport from \\"../macro\\";

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;

exports[`macros with options: with options 1`] = `
"
import loadableImport from \\"../macro\\";
import loadable from \\"loadable-components\\";

const AsyncComponent = loadable(() => loadableImport(\\"./MyComponent\\"), {
  LoadingComponent: () => \\"Loading...\\",
});

      ↓ ↓ ↓ ↓ ↓ ↓

import loadable from \\"loadable-components\\";

const AsyncComponent = loadable(
  () => import(/*webpackChunkName: MyComponent*/ \\"./MyComponent\\"),
  {
    LoadingComponent: () => \\"Loading...\\",
    modules: [\\"MyComponent\\"],
  }
);

"
`;
